addon/uploaders/uploader.js

```diff
@@ -108,14 +108,32 @@ export default EmberObject.extend(Evented, {
   },
 
   /**
-   * Triggers didUpload event with given params and sets isUploading to false
+   * Triggers didUpload and didUpload.withHeaders events with given params
+   * and sets isUploading to false
    *
    * @param {object} data Object of data supplied to the didUpload event
+   * @param {object} status Status of the request
+   * @param {object} xhr XMLHttpRequest instance for this request
    * @return {object} Returns the given data
    */
-  didUpload (data) {
+  didUpload (data, status, xhr) {
     set(this, 'isUploading', false);
     this.trigger('didUpload', data);
+
+    const headers = {};
+    xhr
+      .getAllResponseHeaders().trim().split(/[\r\n]+/)
+      .map((header) => {
+        const [key, value] = header.split(': ');
+        headers[key] = value;
+      }),
```

Comment:
This is not a super candidate for `map` since:
 - it does not return anything
 - it alters the values of something else.

Maybe we could have a `map` (to get key+value) then a reduce:

```js
// Get headers as a key-value object.
const headers = xhr
  .getAllResponseHeaders().trim().split(/[\r\n]+/)
  .map((header) => header.split(': '))
  .reduce((headers, [key, value]) => {
    headers[key] = value;
  }, {});
```

Or `Object.fromEntries`:

```js
// Get headers as a key-value object.
const headers = Object.fromEntries(
  xhr.getAllResponseHeaders().trim().split(/[\r\n]+/)
  .map((header) => header.split(': '))
);
```

Or if you prefer your first approach, you should change to a `forEach`:

```js
// Get headers as a key-value object.
const headers = {};
xhr
  .getAllResponseHeaders().trim().split(/[\r\n]+/)
  .forEach((header) => {
    const [key, value] = header.split(': ');
    headers[key] = value;
  });
```

Warning: don't forget to change the `,` with a `;` line 129

Reply:
Definitely an improper use of `map`.
Went with `Object.entries` option. Seemed clearer to me, even with my love of functional programming :)

Reply:
Moved back to `forEach` as `Object.entries` not supported by project configuration.